keps/sig-network/0013-20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -0,0 +1,438 @@
+         * [kube-proxy Operation](#kube-proxy-operation)
+            * [Kube-Proxy Startup Configuration Changes](#kube-proxy-startup-configuration-changes)
+               * [Multiple bind addresses configuration](#multiple-bind-addresses-configuration)
+               * [Multiple cluster CIDRs configuration](#multiple-cluster-cidrs-configuration)
+         * [IPVS Support and Operation](#ipvs-support-and-operation)
+         * [Health/Liveness/Readiness Probes for Dual-Stack Pods](#healthlivenessreadiness-probes-for-dual-stack-pods)
+         * [CoreDNS Operation](#coredns-operation)
+         * [Ingress Controller Operation](#ingress-controller-operation)
+            * [GCE Ingress Controller: Out-of-Scope, Testing Deferred For Now](#gce-ingress-controller-out-of-scope-testing-deferred-for-now)
+            * [NGINX Ingress Controller - Dual-Stack Support for Bare Metal Clusters](#nginx-ingress-controller---dual-stack-support-for-bare-metal-clusters)
+         * [Load Balancer Operation](#load-balancer-operation)
+         * [Cloud Provider Plugins Considerations](#cloud-provider-plugins-considerations)
+         * [Container Environment Variables](#container-environment-variables)
+         * [Kubeadm Support](#kubeadm-support)
+            * [Kubeadm Configuration Options](#kubeadm-configuration-options)
+            * [Kubeadm-Generated Manifests](#kubeadm-generated-manifests)
+         * [Spf13/pflag](#vendorgithubcomspf13pflag)
+         * [End-to-End Test Support](#end-to-end-test-support)
+         * [User Stories](#user-stories)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Graduation Criteria](#graduation-criteria)
+      * [Implementation History](#implementation-history)
+      * [Alternatives](#alternatives)
+         * [Dual Stack at the Edge](#dual-stack-at-the-edge)
+         * [Variation: Dual-Stack Service CIDRs (a.k.a. Full Dual Stack)](#variation-dual-stack-service-cidrs-aka-full-dual-stack)
+            * [Benefits](#benefits)
+            * [Changes Required](#changes-required)
+
+## Summary
+
+This proposal adds IPv4/IPv6 dual stack functionality to Kubernetes clusters. This includes the following concepts:
+- Awareness of multiple IPv4/IPv6 address assignments per pod
+- Native IPv4-to-IPv4 in parallel with IPv6-to-IPv6 communications to, from, and within a cluster
+
+## Motivation
+
+The adoption of IPv6 has increased in recent years, and customers are requesting IPv6 support in Kubernetes clusters. To this end, the support of IPv6-only clusters was added as an alpha feature in Kubernetes Version 1.9. Clusters can now be run in either IPv4-only, IPv6-only, or in a very limited dual-stack configuration. This limited dual-stack support might be called "headless", in the sense that a CNI network plugin can configure dual-stack addresses on a pod, but Kubernetes remains aware of only a single IP address per pod.  This "headless" dual-stack support is limited by the following restrictions:
```

Comment:
Let's please not use the term "headless" here. It already has a specific meaning in the context of Kubernetes, which is no assigned service IP.